econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -892,30 +991,7 @@ def _safe_result_index(self, X, feature_index):
         (result,) = results
```

Comment:
On line 740 we should not be calling get_inference separately for every attribute, as getting inference can be computationally very costly.

Reply:
Good catch, fixed.